contracts/ExecutionManager.sol

```diff
@@ -67,42 +86,41 @@ contract ExecutionManager is IExecutionManager, AccessControl {
     /**
      * @notice Function for getting the minimum fees required for executing a cross-chain transaction
      * @dev This function is called at source to calculate the execution cost.
-     * @param msgGasLimit_ Gas limit for the transaction
      * @param siblingChainSlug_ Sibling chain identifier
      * @return Minimum fees required for executing the transaction
      */
     function getMinFees(
-        uint256 msgGasLimit_,
         uint32 siblingChainSlug_
     ) external view override returns (uint256) {
-        return _getMinExecutionFees(msgGasLimit_, siblingChainSlug_);
+        return executionFees[siblingChainSlug_];
     }
 
-    /**
-     * @dev Function for getting the minimum fees required for executing a cross-chain transaction
-     * @param msgGasLimit_ Gas limit for the transaction
-     * @param dstChainSlug_ Destination chain identifier
-     * @return Minimum fees required for executing the transaction
-     */
-    function _getMinExecutionFees(
-        uint256 msgGasLimit_,
-        uint32 dstChainSlug_
-    ) internal view returns (uint256) {
-        uint256 dstRelativeGasPrice = gasPriceOracle__.relativeGasPrice(
-            dstChainSlug_
+    function setExecutionFees(
+        uint256 nonce_,
+        uint32 dstChainSlug_,
+        uint256 executionFees_,
+        bytes calldata signature_
+    ) external override {
+        address feesUpdater = signatureVerifier__.recoverSignerFromDigest(
+            keccak256(
+                abi.encode(
+                    FEES_UPDATE_SIG_IDENTIFIER,
```

Comment:
need address(this) here to avoid signature replay on new execution manager deployment on same chain